umpost/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,201 @@
+}
+
+
+def set_esm1p5_fields_file_pattern(run_id):
+    """
+    Generate regex pattern for finding current experiment's UM outputs.
+
+    Parameters
+    ----------
+    run_id : 5 character run ID for the current UM simulation.
+
+    Returns
+    -------
+    fields_file_name_pattern: Regex pattern for matching fields file names
+    """
+
+    # For ESM1.5 simulations, files start with run_id + 'a' (atmosphere) +
+    # '.' (absolute standard time convention) + 'p' (pp file).
+    # See get_name.F90 in the UM7.3 source code for details.
+    fields_file_name_pattern = rf"^{run_id}a.p[a-z0-9]+$"
+
+    return fields_file_name_pattern
+
+
+def set_nc_write_path(fields_file_path, nc_write_dir):
+    """
+    Set filepath for writing NetCDF to based on fields file name.
+
+    Parameters
+    ----------
+    fields_file_path : path to single UM fields file to be converted.
+    nc_write_dir : path to target directory for writing NetCDF files.
+
+    Returns
+    -------
+    nc_write_path : path for writing converted fields_file_path file.
+    """
+    fields_file_path = (
+        Path(fields_file_path)
+        if isinstance(fields_file_path, str)
+        else fields_file_path
+    )
+
+    nc_write_dir = Path(nc_write_dir) if isinstance(nc_write_dir, str) else nc_write_dir
+
+    fields_file_name = fields_file_path.name
+    nc_name = fields_file_name + ".nc"
+    nc_write_path = nc_write_dir / nc_name
+
+    return nc_write_path
+
+
+def find_matching_fields_files(dir_contents, fields_file_name_pattern):
+    """
+    Find files in list of paths with names matching fields_file_name_pattern.
+    Used to find ESM1.5 UM outputs in a simulation output directory.
+
+    Parameters
+    ----------
+    dir_contents : list of file paths, typically contents of a single directory.
+    fields_file_name_pattern : Regex pattern for matching fields file names.
+
+    Returns
+    -------
+    fields_file_paths : subset of dir_contents with names matching fields_file_name_pattern.
+    """
+
+    fields_file_paths = []
+    for filepath in dir_contents:
+        filepath = Path(filepath)
+
+        if re.match(fields_file_name_pattern, filepath.name):
+            fields_file_paths.append(filepath)
+
+    return fields_file_paths
+
+
+def convert_fields_file_list(fields_file_path_list, nc_write_dir):
+    """
+    Convert group of fields files to NetCDF, writing output in nc_write_dir.
+
+    Parameters
+    ----------
+    fields_file_list : list of paths to fields files for conversion.
+    nc_write_dir : directory to save NetCDF files into.
+
+    Returns
+    -------
+    None
+    """
+
+    for fields_file_path in fields_file_path_list:
+
+        fields_file_path = (
+            Path(fields_file_path)
+            if isinstance(fields_file_path, str)
+            else fields_file_path
+        )
+
+        nc_write_path = set_nc_write_path(fields_file_path, nc_write_dir)
+
+        print("Converting file " + fields_file_path.name)
```

Comment:
This `print` is another bit of "buried" I/O. Is this carried over from earlier versions of the run script?

A possible option is modifying this func to return a pair of sequences, one for successful conversions & ones with errors. The higher level calling function can report success/failure to the user. This means tinkering with the test mocks though...

Reply:
I've added a new issue for this in #47